packages/@aws-cdk/core/lib/arn.ts

```diff
@@ -268,3 +284,45 @@ function parseToken(arnToken: string, sep: string = '/', hasName: boolean = true
     return { partition, service, region, account, resource, resourceName, sep };
   }
 }
+
+
+/**
+ * Validate that a string is either unparseable or looks mostly like an ARN
+ */
+function parseArnShape(arn: string): 'unparseable' | string[] {
+  const components = arn.split(':') as Array<string>;
+  const looksLikeArn = arn.startsWith('arn:') && components.length >= 6 && components.length <= 7;
```

Comment:
Nit: I like to use `<` and `>` over `<=` and `>=`. I find the extra "equals" harder to understand on what's included and excluded.

Maybe it's just my head 🤷‍♂️

Reply:
Don't tempt me or I'm going to write 

```
5 < components.length
```
😛